lib/ra10ke/validate.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'ra10ke/monkey_patches'
+require 'tempfile'
+require 'table_print'
+require 'ra10ke/puppetfile_parser'
+require 'English'
+
+module Ra10ke
+  module Validate
+
+    GOOD_EMOJI = ENV['GOOD_EMOJI'] || '👍'
+    BAD_EMOJI = ENV['BAD_EMOJI'] || '😨'
+
+    # Validate the git urls and refs
+    def define_task_validate(*args)
+      desc 'Validate the git urls and branches, refs, or tags'
+      task :validate do
+        gitvalididation = Ra10ke::Validate::Validation.new(puppetfile_path)
```

Comment:
Using `puppetfile_path` will not work in all cases. If someone only configures `basepath` it will be nil but the Puppetfile will be in a different location. You could use `get_puppetfile.puppetfile_path` instead.

Reply:
Updated to use this method.